src/commands.ts

```diff
@@ -0,0 +1,168 @@
+    let project = '';
+    if (!args) {
+        const activeEditor = vscode.window.activeTextEditor;
+        if (activeEditor !== undefined) {
+            project = activeEditor.document.uri.fsPath;
+        }
+    } else {
+        project = args.fsPath;
+    }
+
+    commandObserver.clear();
+
+    let isValid = await validateProjectSDK([project], commandObserver);
+    if (!isValid) {
+        return;
+    }
+    await dotnetBuild(dotNetSdk, project, commandObserver, cancelToken);
+}
+
+async function buildAllProjects(dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
+    try {
+        let projects = await vscode.workspace.findFiles('{**/*.pgproj}');
+        commandObserver.clear();
+
+        let isValid = await validateProjectSDK(projects.map(p => p.fsPath), commandObserver);
+        if (!isValid) {
+            return;
+        }
+
+        for (let project of projects) {
+            if (cancelToken.isCancellationRequested) {
+                return;
+            }
+            await dotnetBuild(dotNetSdk, project.fsPath, commandObserver, cancelToken);
+        }
+    } catch (err) {
+        vscode.window.showErrorMessage(err);
+    }
+}
+
+async function validateProjectSDK(projects: string[], commandObserver: CommandObserver): Promise<boolean> {
+    var packageInfo = Utils.getPackageInfo();
+    let unsupportedProjects = await Helper.checkProjectVersion(packageInfo.minSupportedPostgreSQLProjectSDK, packageInfo.maxSupportedPostgreSQLProjectSDK, projects, commandObserver);
+    if (unsupportedProjects && unsupportedProjects.length > 0) {
+        unsupportedProjects.map(p => commandObserver.logToOutputChannel(strings.format(Constants.buildFailedUnsupportedSdkMessage, p)));
+        return Promise.resolve(false);
+    }
+    return Promise.resolve(true);
+}
+
+async function dotnetBuild(dotNetSdk: DotNetInfo, project: string, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
+    let args = ['build', project];
+    commandObserver.logToOutputChannel(strings.format(Constants.buildStartedMessage, project));
+    await runDotNetCommand(dotNetSdk, args, commandObserver, cancelToken);
+    commandObserver.logToOutputChannel(strings.format(Constants.buildCompletedMessage, project));
+}
+
+async function addNewPostgreSQLProject(args: vscode.Uri, dotNetSdk: DotNetInfo, projectTemplateNugedId: string, commandObserver: CommandObserver) {
+    let folder = args.fsPath;
+    var defaultProjectName = path.basename(folder);
+
+    var projectName = await vscode.window.showInputBox(
+        {
+            prompt: 'Project Name',
+            value: defaultProjectName,
+            validateInput: (value: string) => {
+                if (!value || value.trim().length === 0) {
+                    return Constants.projectNameEmptyErrorMessage;
```

Comment:
Are these constants localizable? I know for DAMT we used some ```vscode-nls``` package

Reply:
Currently they are not. Yes, would have to use that package. As everything is in Constants, it would be easy.